sktime/forecasting/model_selection/_split.py

```diff
@@ -97,11 +98,35 @@ def has_changed(k, v):
 
 
 def _check_y(y):
-    """Check input to `split` function."""
+    """Coerce input to `split` function.
+
+    Parameters
+    ----------
+    y : pd.Series, pd.DataFrame, np.ndarray, or pd.Index
+
+    Returns
+    -------
+    check_time_index(y_index), where y_index is as follows:
+        if y is pd.Series, pd.DataFrame, y_index = y.index
+        if y is pd.Index, y_index = y itself
+        if y is np.ndarray, y_index =  pd.Index(y)
+
+    Raises
+    ------
+    TypeError if y is not of one of the expected types
+    NotImplementedError if y_index is not a supported sktime index type
+    ValueError if y_index is not monotonous
```

Comment:
```
TypeError 
    if y is not of one of the expected types
NotImplementedError 
    if y_index is not a supported sktime index type
ValueError 
    if y_index is not monotonous
```
I know you've only got one case for each but my ocd is getting the better of me here.

Reply:
done